pkg/odo/cli/component/create.go

```diff
@@ -368,18 +355,18 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 			// Component namespace: User needs to specify component namespace,
 			// by default it is the current active namespace if it can't get from --project flag or --namespace flag
-			if len(co.devfileMetadata.componentNamespace) == 0 && !pushtarget.IsPushTargetDocker() {
-				if cmd.Flags().Changed("project") {
-					componentNamespace, err = cmd.Flags().GetString("project")
-					if err != nil {
-						return err
-					}
-				} else {
-					componentNamespace = ui.EnterDevfileComponentNamespace(defaultComponentNamespace)
+			if co.devfileMetadata.componentNamespace == "" && !pushtarget.IsPushTargetDocker() {
```

Comment:
Is there a potential to convert this into a function? Asking because I see a repetition later in the file